src/store/reducers/movies.ts

```diff
@@ -0,0 +1,200 @@
+            state.data = action.payload
+        },
+        setShowHeader: (state, action: PayloadAction<boolean>): void => {
+            state.isShowHeader = action.payload
+        },
+        setOpenForm: (state, action: PayloadAction<boolean>): void => {
+            state.isOpenForm = action.payload
+        },
+        setEditStatus: (state, action: PayloadAction<boolean>): void => {
+            state.isEdit = action.payload
+        },
+        setMovie: (state, action: PayloadAction<Movie>): void => {
+            state.data[action.payload.key] = action.payload.value
+        }
+    }
+})
+const movies = createSlice({
+    name: 'movies',
+    initialState: {
+        data: [],
+        total: 0,
+        offset: 0,
+        totalAmount: 0
+    } as IMoviesState,
+    reducers: {
+        addMoviesState: (state, action: PayloadAction<IMoviesState>): IMoviesState => {
+            return action.payload
+        },
+        addMovies: (state, action: PayloadAction<MoviesList>): void => {
+            state.data = action.payload
+        },
+        pushMovie: (state, action: PayloadAction<Movie>): void => {
+            state.data.push(action.payload)
+        },
+        putMovie: (state, action): void => {
+            const id = action.payload.id
+            const index = state.data.findIndex((movie) => movie.id == id)
+            
+            if(index != -1) {
+                state.data.splice(index, 1, action.payload)
+            }
+        },
+        sortMovies: (state, action: PayloadAction<string>): void => {
+            switch(action.payload) {
+                case 'date':
+                    const getTime = (date: Date) => {
+                        return date != null ? new Date(date).getTime() : 0;
+                    }
+                    state.data.sort((a: any, b: any) => {
+                        return getTime(b.release_date) - getTime(a.release_date);
+                    })
+                    break
+                default:
+                    state.data.sort((a: any, b: any) => {
+                        return b.vote_average - a.vote_average;
+                    })
+                    break
+            }
+            
+        }
+    }
+})
+export const { addMoviesState, addMovies, pushMovie, putMovie, sortMovies  } = movies.actions
+export const { addMovie, setShowHeader, setOpenForm, setEditStatus, setMovie } = movie.actions
+
+export const selectMovie = (state: RootState) => state.movie.data
+export const selectShowHeader = (state: RootState) => state.movie.isShowHeader
+export const selectOpenForm = (state: RootState) => state.movie.isOpenForm
+export const selectEditStatus = (state: RootState) => state.movie.isEdit
+
+export const selectListGenres = (state: RootState) => {
+    const list = []
+    const rs = state.movies.data.map((movie: any) => {
+        return movie.genres
+    })
+    return [...new Set(list.concat.apply([], rs))]
```

Comment:
list is always empty array. Why do you need this? Could you write smth like this?
```js
return [...new Set([...rs]))]
```